pytensor/tensor/slinalg.py

```diff
@@ -64,21 +65,52 @@ def make_node(self, x):
         dtype = scipy_linalg.cholesky(np.eye(1, dtype=x.type.dtype)).dtype
         return Apply(self, [x], [tensor(shape=x.type.shape, dtype=dtype)])
 
+    def _cholesky(
+        self, a, lower=False, overwrite_a=False, clean=True, check_finite=False
+    ):
+        a1 = np.asarray_chkfinite(a) if check_finite else np.asarray(a)
+
+        # Squareness check
+        if a1.shape[0] != a1.shape[1]:
+            raise ValueError(
+                "Input array is expected to be square but has "
+                f"the shape: {a1.shape}."
+            )
+
+        # Quick return for square empty array
+        if a1.size == 0:
+            dt = self._cholesky(np.eye(1, dtype=a1.dtype)).dtype
+            return np.empty_like(a1, dtype=dt), lower
+
+        overwrite_a = overwrite_a or _datacopied(a1, a)
+        (potrf,) = scipy_linalg.get_lapack_funcs(("potrf",), (a1,))
+        c, info = potrf(a1, lower=lower, overwrite_a=overwrite_a, clean=clean)
+        if info > 0:
+            raise scipy_linalg.LinAlgError(
+                f"{info}-th leading minor of the array is not positive definite"
+            )
+        if info < 0:
+            raise ValueError(
+                f"LAPACK reported an illegal value in {-info}-th argument "
+                f'on entry to "POTRF".'
+            )
```

Comment:
this depends `on_error` flag

Reply:
I think the behavior was the same because of the try/catch, but it was clunky. It is simpler after refactoring out the helper.